modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/AbstractPageMemoryMvPartitionStorage.java

```diff
@@ -1000,16 +1007,18 @@ public void close() {
         busyLock.block();
 
         try {
-            IgniteUtils.closeAll(getResourcesToClose());
+            IgniteUtils.closeAll(getResourcesToClose(goingToDestroy));
         } catch (Exception e) {
             throw new StorageException(e);
         }
     }
 
     /**
      * Returns resources that should be closed on {@link #close()}.
+     *
+     * @param goingToDestroy If the closure is in preparation for destruction.
      */
-    protected List<AutoCloseable> getResourcesToClose() {
+    protected List<AutoCloseable> getResourcesToClose(boolean goingToDestroy) {
```

Comment:
Why do you need the boolean flag here?

Reply:
This comment explains the matter: https://github.com/apache/ignite-3/pull/1559#discussion_r1083726707